### PR TITLE
allow running mesh-announce with sudo

### LINKED ISSUE #62: allow running mesh-announce as non root
I would like to run mesh-announce as a user with less permissions then root.

For this to work, as far as I understand it, I need two things::

- allow the "mesh-announce" user to use ports under 1024: 
  https://github.com/Freifunk-Rhein-Neckar/salt/blob/01a7299dd17d76bed5639fd52ff23f9db63ce577/mesh-announce/files/mesh-announce.service.j2#L10-L11
- enable mesh-announce to run `batctl` commands without being root. 
  I was thinking about a sudo wrapper and only allowing the batctl command for that user
  `batctl` is called two times:
  - https://github.com/ffnord/mesh-announce/blob/acee4362d4e8da9edd5bb1b3a63d9182325d4f5a/providers/nodeinfo/network/mesh.py#L16
  - https://github.com/ffnord/mesh-announce/blob/acee4362d4e8da9edd5bb1b3a63d9182325d4f5a/providers/neighbours/batadv.py#L11
 
How bad would it be to put a `sudo` in front of the `batctl` call? Or would we need detection if `sudo` is available, or an config option, ... Whats the best approach for a PR to get accepted? :wink:

### PR DESCRIPTION
**Pull request by herbetom**
closes #62

based on structure of #51

### CODE REVIEW

File: respondd.service [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 Description=Respondd
 After=network.target
 
 [Service]
 ExecStart=/opt/mesh-announce/respondd.py -d /opt/mesh-announce/providers -i <your-clientbridge-if> -i <your-mesh-vpn-if> -b <your-batman-if> -m <mesh ipv4 address>
 Restart=always
+WorkingDirectory=/opt/mesh-announce
 Environment=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+# User=mesh-announce
+# AmbientCapabilities=CAP_NET_BIND_SERVICE
```
Comment (AiyionPrime) [inline respondd.service:12]: This needs some documentation, anywhere. If not in the main-readme, then in a 'drop priviliges'-readme.

File: providers/util.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
     lines = output.splitlines()
     lines = [line.decode("utf-8") for line in lines]
     return lines
+
+def call_batctl(batadv_dev, args):
+
+    if os.geteuid() > 0:
+        return call(['sudo', 'batctl', 'meshif', batadv_dev] + args)
+
+    return call(['batctl', '-m', batadv_dev] + args)
```
Comment (AiyionPrime) [inline providers/util.py:15]: This needs refactoring and a rebase after the merge of #51 or #64.

Comment (AiyionPrime) [reply providers/util.py:15]: As soon as it does, one should write the list into a var in order not to repeat it and insert sudo at index zero before returning it.
Possibly like this:

```python3
def call_batctl(batadv_dev, args):
    base_args = ['batctl', 'meshif', batadv_dev]
    if os.geteuid() > 0:
        base_args.insert(0, 'sudo')
    return call(base_args + args)
```

File: respondd.service [hunk 3]
```diff
@@ -1,11 +1,15 @@
+
```
Comment (AiyionPrime) [inline respondd.service:1]: Is this intended?

**Review by AiyionPrime — APPROVED** [review]
This looks fine but conflicts #51 and #64 and should be merged after one of them was merged, imho.

File: respondd.service (hunk 1 above)
Comment (herbetom) [reply respondd.service:12]: Yes, that needs to be documented. But I'm not so sure what the best general approach would be. 

Here is how i do it at the moment: https://github.com/Freifunk-Rhein-Neckar/salt/blob/master/mesh-announce/init.sls

File: providers/util.py (hunk 2 above)
Comment (herbetom) [reply providers/util.py:15]: Yes, this conflicts with #51, I created #80 (replacing #51) with which the conflict should be smaller.

Comment (herbetom) [reply providers/util.py]: I implemented it with the list both here and in #80

File: respondd.service (hunk 3 above)
Comment (herbetom) [reply respondd.service]: no, i fixed that

**Pull request merged**